src/Http/Controllers/ImportController.php

```diff
@@ -37,46 +36,67 @@ public function preview(NovaRequest $request, $file)
+        $resources = $resources->mapWithKeys(function ($resource) {
+            return [$resource::uriKey() => $resource::label()];
+        });
 
-            if(!isset($static_vars['canImportResource'])) {
-                return true;
-            }
+        return response()->json(compact('sample', 'resources', 'fields', 'total_rows', 'headings'));
+    }
 
-            return isset($static_vars['canImportResource']) && $static_vars['canImportResource'];
-        });
+    public function getAvailableFieldsForImport(String $resource, $request)
+    {
+        $novaResource = new $resource(new $resource::$model);
+        $fieldsCollection = collect($novaResource->creationFields($request));
 
-        $fields = $resources->map(function ($resource) {
-            $model = $resource::$model;
+        if (method_exists($novaResource, 'exceptAttributesImportResource')) {
+            $fieldsCollection = $fieldsCollection->filter(function(Field $field) use ($novaResource, $request){
+                return !in_array($field->attribute, $novaResource->exceptAttributesImportResource($request));
+            });
+        }
 
-            return new $resource(new $model);
-        })->mapWithKeys(function (Resource $resource) use ($request) {
-            $fields = collect($resource->creationFields($request))
-                ->map(function (Field $field) {
+        $fields = $fieldsCollection->map(function (Field $field) {
                     return [
                         'name' => $field->name,
                         'attribute' => $field->attribute
                     ];
                 });
-            return [$resource->uriKey() => $fields];
-        });
+        
+       return [$novaResource->uriKey() => $fields];
+    }
 
-        $resources = $resources->mapWithKeys(function ($resource) {
-            return [$resource::uriKey() => $resource::label()];
-        });
+    public function getAvailableResourcesForImport(NovaRequest $request) {
 
-        return response()->json(compact('sample', 'resources', 'fields', 'total_rows', 'headings'));
+        $novaResources = collect(Nova::authorizedResources($request));
+
+        return $novaResources->filter(function ($resource) use ($request) {
+                    if ($resource === ActionResource::class) {
+                        return false;
+                    }
+
+                    if (!isset($resource::$model)) {
+                        return false;
+                    }
+                    
+                    $resourceReflection = (new \ReflectionClass((string) $resource));
+                    
+                    if($resourceReflection->hasMethod('canImportResource')) {
+                        return $resource::canImportResource($request);
+                    }
+
+                    $static_vars = $resourceReflection->getStaticProperties();
+
+                    if(!isset($static_vars['canImportResource'])) {
```

Comment:
Minor code style change

```suggestion
                    if (!isset($static_vars['canImportResource'])) {
```